packages/ui/src/components/va-stepper/VaStepper.vue

```diff
@@ -0,0 +1,200 @@
+    >
+      <template
+        v-for="step, i in $props.steps"
+        :key="i"
+      >
+        <div
+          class="va-stepper__step-content"
+          v-if="$slots[`step-content-${i}`] && modelValue === i"
+        >
+          <slot
+            :name="`step-content-${i}`"
+            v-bind="{ ...stepControls, step }"
+          />
+        </div>
+      </template>
+      <div class="va-stepper__controls">
+        <va-stepper-controls
+          v-if="!controlsHidden"
+          v-bind="{ ...$props, stepControls }"
+        />
+        <slot
+          name="controls"
+          v-bind="stepControls"
+        />
+      </div>
+    </div>
+  </div>
+</template>
+<script lang="ts">
+import { defineComponent, PropType, Ref } from 'vue'
+import { useColors, useStateful, useStatefulProps } from '../../composables'
+import type { Step, StepControls } from './types'
+import VaStepperControls from './VaStepperControls.vue'
+import VaStepperStepButton from './VaStepperStepButton.vue'
+
+export default defineComponent({
+  name: 'VaStepper',
+  components: { VaStepperControls, VaStepperStepButton },
+  props: {
+    ...useStatefulProps,
+    modelValue: { type: Number, default: 0 },
+    steps: {
+      type: Array as PropType<Step[]>,
+      default: () => [],
+      required: true,
+    },
+    color: { type: String, default: 'primary' },
+    vertical: { type: Boolean, default: false },
+    navigationDisabled: { type: Boolean, default: false },
+    controlsHidden: { type: Boolean, default: false },
+    nextDisabled: { type: Boolean, default: false },
+  },
+  emits: ['update:modelValue'],
+  setup (props, { emit }) {
+    const { valueComputed: modelValue }: { valueComputed: Ref<number> } = useStateful(props, emit, 'modelValue', { defaultValue: 0 })
+
+    const { getColor } = useColors()
+    const stepperColor = getColor(props.color)
+
+    const isNextStepDisabled = (index: number) => props.nextDisabled && index > modelValue.value
+
+    const setStep = (index: number) => {
+      if (props.steps[index].disabled) { return }
+      emit('update:modelValue', index)
+    }
+
+    const nextStep = (stepsToSkip = 0) => {
+      const targetIndex = modelValue.value + 1 + stepsToSkip
+
+      if (!props.steps[targetIndex]) { return }
+      if (props.steps[targetIndex].disabled) {
+        nextStep(stepsToSkip + 1)
+      }
+
+      setStep(targetIndex)
+    }
+
+    const prevStep = (stepsToSkip = 0) => {
+      const targetIndex = modelValue.value - 1 - stepsToSkip
+
+      if (!props.steps[targetIndex]) { return }
+      if (props.steps[targetIndex].disabled) {
+        prevStep(stepsToSkip + 1)
+      }
+
+      setStep(targetIndex)
+    }
+
+    const stepControls: StepControls = { setStep, nextStep, prevStep }
+
+    return {
+      isNextStepDisabled,
+      stepperColor,
+      getColor,
+      stepControls,
+    }
+  },
+})
+</script>
+
+<style lang="scss">
+@import "../../styles/resources";
```

Comment:
What's the purpose of import this?

Reply:
I actually don't know. I used other component as reference and they all seem to have it. Should I remove it?

Reply:
That's minor. You can remove it, @LighthouseKeeperYN.